static/app/utils/onDemandMetrics/index.tsx

```diff
@@ -14,12 +14,21 @@ function isOnDemandSupportedFilterKey(key: FieldKey): boolean {
   return ON_DEMAND_METRICS_SUPPORTED_TAGS.has(key);
 }
 
+function isCustomTag(key: FieldKey): boolean {
+  return !getFieldDefinition(key);
+}
+
 export function createOnDemandFilterWarning(warning: React.ReactNode) {
-  return (key: string) =>
-    !isStandardSearchFilterKey(key as FieldKey) &&
-    isOnDemandSupportedFilterKey(key as FieldKey)
-      ? warning
-      : null;
+  return (key: string) => {
+    const fieldKey = key as FieldKey;
+    if (isCustomTag(fieldKey)) {
+      return warning;
+    }
+    if (!isStandardSearchFilterKey(fieldKey) && isOnDemandSupportedFilterKey(fieldKey)) {
+      return warning;
+    }
```

Comment:
Just to make sure, so there is the case that the key is not standard not custom but also not supported for onDemand.
We cannot simply do
```ts
return !isStandardSearchFilterKey(fieldKey) ? warning : null
```
right?

Reply:
Yes, and there are also keys that are standard but not supported by on demand